From linkml to schemapack

sbilge opened the pull request:
This PR includes scripts and files to convert ghga-metadata-model from its LinkML representation to SchemaPack definition. 
The bash script `migrate_to_schemapack.sh` calls the scripts in the required order and cleans up the repository from LinkML-related files and folders.

dontseyit commented on scripts/extract_class_content.py:
```diff
@@ -0,0 +1,163 @@
+def linkml_to_json(file: Path) -> dict:
+    """Create a JSON representation of the schema described in LinkML using LinkML
+    command line option."""
+    with subprocess.Popen(["gen-json-schema", file], stdout=subprocess.PIPE) as process:
+        stdout, _ = process.communicate()
+    return json.loads(stdout.decode("utf-8"))
```
You may want to add some error handling here for `subprocess.CalledProcessError` when command fails and `json.JSONDecodeError` when json decoding fails

sbilge replied:
Thanks for the suggestion, but it's okay to leave it as is. This script won't be part of the repository—it's just a helper to convert the LinkML to SchemaPack, and it won't be needed once we complete the transition. Also, the script wouldn't work with the current `requirements.txt` file anyway since we've already removed all the LinkML-related packages. It is a part of the PR for archiving purposes :)

dontseyit commented on scripts/extract_relationships.py:
```diff
@@ -0,0 +1,115 @@
+
+def resolve_inherited_relations(schema: Schema) -> Schema:
+    """Resolves the relations from the class inheritances"""
+    for _key, value in schema.classes.items():
+        if value.inheritance and schema.classes.get(value.inheritance):
+            inherited_relations = schema.classes[value.inheritance].relations
+            value.relations.update(inherited_relations)
```
Better readability and more clarity when definitive variable names are used:

```suggestion
    for schema_class in schema.classes.values():
        if schema_class.inheritance and schema_class.inheritance in schema.classes:
            inherited_relations = schema.classes[schema_class.inheritance].relations
            schema_class.relations.update(inherited_relations)
```

dontseyit reviewed (dismissed):
I looks good to me, I'm not able to review schemapack logic at the moment but I suggest some improvements to the code.

sbilge commented:
I updated the branch based on the ghga-metadata-schema (on linkml) changes. I introduced a new condition to  `extract_relationships.py` script to prevent the script from interpreting two specific fields as a relationship. The explanation is also provided in the function definition of `slots_with_class_range`,
